statsd/statsdtest/recorder.go

```diff
@@ -0,0 +1,85 @@
+package recorder
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/falun/go-statsd-client/statsd"
+)
+
+// RecordingSender implements statsd.Sender but parses individual Stats into a
+// buffer that can be later inspected instead of sending to some server. It
+// should constructed with NewRecordingSender().
+type RecordingSender struct {
+	m      sync.Mutex
+	buffer Stats
+	closed bool
+}
+
+var _ statsd.Sender = &RecordingSender{}
```

Comment:
Exactly. I can push it into a test if you're more comfortable with that.

The type assertion check won't work because I need to verify that a non-interface type meets an interface (as opposed to checking if an interface is a given type). As such I'll be sticking with the `var _ statsd.Sender` approach unless I misunderstood your suggestion.